### PR TITLE
Rework navigation tool hints and navigation shortcuts

### LINKED ISSUE #1352: Dropdown button for rotating camera
Currently the viewport can be rotated by holding <kbd>Ctrl</kbd> whilst dragging with the middle mouse button. This is however difficult to find. A similar option should be added to the menu bar at the top of the app, in the view menu to rotate the canvas.

The code for the menu bar can be found in `editor/src/messages/portfolio/menu_bar/menu_bar_message_handler.rs:250`.
The code for the canvas rotation can be found in `editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:197`.

Inspiration can be taken from similar views in inkscape and illustrator 
inkscape camera orientation dropdownillustrator camera rotate dropdown

### LINKED ISSUE #1365: Rearrange Pan/Tilt/Zoom features for better discoverability
- Switch Ctrl+MMB to Alt+MMB for tilt from any tool
- Improve the layout of the View menu
- More (ask Keavon)

### PR DESCRIPTION
**Pull request by mobile-bungalow**
Wrap-up of https://github.com/GraphiteEditor/Graphite/pull/1364

Closes #1352
Closes #1365

Notes: There might be a little more action toggling that you expect in this PR. The  hotkeys for one-mouse-button navigation conflicted with the node overlay toggle and the navigation message handler needed a way to lock the space modifier. additionally, the navigation message handler needed a way to catch mouse up events when dispatched from a menu outside the viewport. This feels like the best way to achieve this without spreading navigation state to tools - even if it's a little ugly.

### CODE REVIEW

File: editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs [hunk 1]
```diff
@@ -151,7 +159,7 @@ impl MessageHandler<NavigationMessage, (&Document, Option<[DVec2; 2]>, &InputPre
 					responses.add(SetCanvasRotation { angle_radians: self.tilt + rotation });
 				}
 
-				if self.zooming {
+				if self.zooming && !self.panning && !self.tilting {
```
Comment (0HyperCube) [inline editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:162]: Perhaps this should be an enum:
```rs
pub enum TransformOperation {
    None,
    Pan,
    Rotate,
    Zoom
}
```

File: editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs [hunk 2]
```diff
@@ -19,35 +19,43 @@ use serde::{Deserialize, Serialize};
...
+			pre_commit_tilt: 0.,
 			tilting: false,
 
 			zoom: 1.,
+			pre_commit_zoom: 0.,
 			zooming: false,
 			snap_zoom: false,
 
 			mouse_position: ViewportPosition::default(),
+			dispatched_from_menu: false,
```
Comment (0HyperCube) [inline editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:58]: I'm not sure having this as state for the message handler - it seems like it should just be for each message?

**Review by 0HyperCube — CHANGES_REQUESTED** [review]
Looks good.

File: editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs (hunk 1 above)
Comment (mobile-bungalow) [reply editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:162]: agreed, I found it like this and stuck to chesterton's fence.

File: editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs (hunk 2 above)
Comment (mobile-bungalow) [reply editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:58]: unfortunately I have to use this in bool in the actions function to enable this to take priority over the currently active tool when starting a translation from outside the viewport. Is there some other way I could thread the state to the action function?

Comment (0HyperCube) [reply editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:58]: I see.

File: editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs (hunk 1 above)
Comment (mobile-bungalow) [reply editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs]: change made. I moved all the transform operation state into the enum, this way invalid states are harder to represent.

File: editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs [hunk 3]
```diff
@@ -15,39 +15,35 @@ use graphene_core::renderer::format_transform_matrix;
...
 	pub tilt: f64,
-	tilting: bool,
-
 	pub zoom: f64,
-	zooming: bool,
-	snap_zoom: bool,
 
+	transform_operation: TransformOperation,
 	mouse_position: ViewportPosition,
+	dispatched_from_menu: bool,
```
Comment (0HyperCube) [inline editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:34]: Thoughts on renaming this to something like `finish_operation_with_click`?

**Review by 0HyperCube — APPROVED** [review]

File: editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs (hunk 3 above)
Comment (mobile-bungalow) [reply editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:34]: maybe a little vague? it finished with a click no matter where it is dispatched from. `blocking_mouse_events` be more specific, it's what it's doing not why it's doing it.

Comment (0HyperCube) [reply editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:34]: It doesn't really finish with a click if you are dragging the middle mouse button - it finishes with mouse up.

Comment (mobile-bungalow) [reply editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs:34]: okay I see your point I'll change it.

**Pull request merged**